### PR TITLE
Fix Timeout handler logging inconsistent issue.

### PR DESCRIPTION
**Pull request by nuwanw**
Fixes #wso2/product-ei/issues/5360

### CODE REVIEW

**Review by malakaganga — APPROVED** [review]

**Review by abeykoon — APPROVED** [review]

File: modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java [hunk 1]
```diff
@@ -181,12 +181,19 @@ public void receive(MessageContext messageCtx) throws AxisFault {
                 messageCtx.removeProperty(PassThroughConstants.INTERNAL_EXCEPTION_ORIGIN);
                 org.apache.synapse.MessageContext SynapseOutMsgCtx = callback.getSynapseOutMsgCtx();
                 ConcurrencyThrottlingUtils.decrementConcurrencyThrottleAccessController(SynapseOutMsgCtx);
-
+                boolean dropMessage = false;
```
Comment (isudana) [inline modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java:184]: can we please change this dropMessage variable name to a more meaningful one?

File: modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java [hunk 2]
```diff
@@ -181,12 +181,19 @@ public void receive(MessageContext messageCtx) throws AxisFault {
...
-                        return;
+                        //callback expired by the timeout handler
+                        dropMessage = true;
+                    } else {
+                        callback.setMarkedForRemoval();
                     }
-                    callback.setMarkedForRemoval();
+                }
+                //callback expired by timeout handler, hence dropping the message without proceed further
+                if(dropMessage) {
```
Comment (isudana) [inline modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java:194]: please fix the formatting issue

File: modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java [hunk 3]
```diff
@@ -201,12 +208,7 @@ public void receive(MessageContext messageCtx) throws AxisFault {
                 }
             } else {
                 // TODO invoke a generic synapse error handler for this message
-                if (!PassThroughConstants.INTERNAL_ORIGIN_ERROR_HANDLER
-                        .equals(messageCtx.getProperty(PassThroughConstants.INTERNAL_EXCEPTION_ORIGIN))) {
-                    log.warn("Synapse received a response for the request with message Id : " + messageID
-                            + " and correlation_id : " + messageCtx.getProperty(CorrelationConstants.CORRELATION_ID)
-                            + " But a callback is not registered (anymore) to process " + "this response");
-                }
+                logWarnNoCallbackRegistered(messageID, messageCtx);
```
Comment (isudana) [inline modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java:211]: can we please rename this method name to something like "handleNoCallbackRegistered"

**Review by isudana — CHANGES_REQUESTED** [review]

File: modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java (hunk 1 above)
Comment (nuwanw) [reply modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java]: renamed to isMarkedForRemoval

File: modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java (hunk 2 above)
Comment (nuwanw) [reply modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java]: Done

File: modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java (hunk 3 above)
Comment (nuwanw) [reply modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java]: renamed to handleNoCallback

**Review by isudana — APPROVED** [review]

**Pull request merged**